modules/microshift-verify-container-signatures-sigstore.adoc

```diff
@@ -0,0 +1,90 @@
+[id="microshift-verify-container-signatures-sigstore_{context}"]
+= Verifying container signatures using sigstore
+
+Verify container signatures for {microshift-short} by configuring the container runtime to use sigstore. The container signature verification uses the public key from the Red Hat keypair when signing the images.
+
+You can access Red Hat public keys at the following link:
+
+* link:https://access.redhat.com/security/team/key[Product Signing Keys]
+
+The release key 3 is used in the following example.
+//Q: should we also be using key2, see "This key is used in addition to release key 2 for signing Red Hat products released after October 2024 and their updates."
+
+.Prerequisites
+
+* You have admin access to the {microshift-short} host.
+
+.Procedure
+
+. Download the relevant public key and save it as `/etc/containers/RedHat_ReleaseKey3.pub` by running the following command:
++
+[source,terminal]
+----
+$ sudo curl -sL https://access.redhat.com/security/data/63405576.txt -o /etc/containers/RedHat_ReleaseKey3.pub
+----
+
+//Q: where did we get this file? does the release key generate it?
+. To configure the container runtime to verify images from Red Hat sources, edit the `/etc/containers/policy.json` file to contain the following configuration:
+
+.Example policy JSON file
+[source,json]
+----
+{
+    "default": [
+        {
+            "type": "reject"
+        }
+    ],
+    "transports": {
+        "docker": {
+            "quay.io/openshift-release-dev": [{
+                "type": "sigstoreSigned",
+                "keyPath": "/etc/containers/RedHat_ReleaseKey3.pub",
+                "signedIdentity": {
+                    "type": "matchRepoDigestOrExact"
+                }
+            }],
+            "registry.redhat.io": [{
+                "type": "sigstoreSigned",
+                "keyPath": "/etc/containers/RedHat_ReleaseKey3.pub",
+                "signedIdentity": {
+                    "type": "matchRepoDigestOrExact"
+                }
+            }]
+        }
+    }
+}
+----
+
+. Configure your remote registries to use sigstore attachments when pulling images to the local storage, by editing the `/etc/containers/registries.d/registry.quay.io.yaml` file to contain the following configuration:
++
+[source,terminal]
+----
+$ cat /etc/containers/registries.d/registry.redhat.io.yaml
+docker:
+     registry.redhat.io:
+         use-sigstore-attachments: true
+----
+
+. Configure your remote registries to use sigstore attachments when pulling images to the local storage, by editing the `/etc/containers/registries.d/registry.redhat.io.yaml` file to contain the following configuration:
++
+[source,terminal]
+----
+$ cat /etc/containers/registries.d/quay.io.yaml
+docker:
+  quay.io/openshift-release-dev:
+    use-sigstore-attachments: true
+----
+
+. Create user-specific registry configuration files if your use case requires signature verification for those image sources.
+//Q: link or guidance of some sort here?
```

Comment:
The contents template is kind of demonstrate above. We can refer to these and it should be enough, I think.